### PR TITLE
cli(ticdc) add checks on pdAddrs in cdc cli

### PR DESCRIPTION
**Pull request by crelax**
### What problem does this PR solve?

Issue Number: close #xxx

#### Tests 

 - Unit test

### Release note 

```
None.
```

### CODE REVIEW

File: pkg/cmd/factory/factory_impl.go [hunk 1]
```diff
@@ -148,7 +153,17 @@ func (f factoryImpl) PdClient() (pd.Client, error) {
 	}
 
 	pdAddr := f.GetPdAddr()
+	if len(pdAddr) == 0 {
+		return nil, cerror.ErrInvalidServerOption.
+			GenWithStack("empty PD address, please use --pd to specify PD cluster addresses")
+	}
 	pdEndpoints := strings.Split(pdAddr, ",")
+	tlsEnabled := len(credential.KeyPath) != 0
```
Comment (liuzix) [inline pkg/cmd/factory/factory_impl.go:161]: Please reuse available method to check whether TLS is enabled.

File: pkg/cmd/factory/factory_impl_test.go [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+	"github.com/golang/mock/gomock"
+	cmdcontext "github.com/pingcap/tiflow/pkg/cmd/context"
+	mock_factory "github.com/pingcap/tiflow/pkg/cmd/factory/mock"
+	"github.com/pingcap/tiflow/pkg/security"
+	"github.com/stretchr/testify/require"
+	"google.golang.org/grpc"
+)
+
+func TestFactoryImpl_PdClient(t *testing.T) {
```
Comment (liuzix) [inline pkg/cmd/factory/factory_impl_test.go:28]:
```suggestion
func TestFactoryImplPdClient(t *testing.T) {
```

**Review by sdojjy — APPROVED** [review]

**Pull request merged**